tests/results/test_aggregator.py

```diff
@@ -503,3 +514,84 @@ def test_read_history_matching_result_data(
     # Receiving an invalid History Matching type should raise.
     with pytest.raises(ValueError, match="type `foobar` not supported"):
         read_history_matching_result(metadata, "foobar")  # type: ignore
+
+
+def test_read_uncertainty_propagation_results(
+    datadir: Path, up_results_dir: Path
+) -> None:
+    empty_metadata = read_uncertainty_propagation_analyses_meta_data(
+        result_directory=datadir
+    )
+    assert empty_metadata.items == {}
+    result = read_uncertainty_propagation_results(
+        metadata=empty_metadata, results_key="absolute_pressure@trend_id_1"
+    )
+    assert result is None
+
+    metadata = read_uncertainty_propagation_analyses_meta_data(
+        result_directory=up_results_dir
+    )
+    assert list(metadata.items.keys()) == [
+        "temperature@trend_id_1",
+        "absolute_pressure@trend_id_1",
+    ]
+
+    assert metadata.items["temperature@trend_id_1"].result_index == 0
+    assert metadata.items["temperature@trend_id_1"].sample_indexes == [
+        [0, 0],
+        [0, 1],
+        [0, 2],
+        [0, 3],
+        [0, 4],
+    ]
+
+    assert metadata.items["absolute_pressure@trend_id_1"].result_index == 1
+    assert metadata.items["absolute_pressure@trend_id_1"].sample_indexes == [
+        [1, 0],
+        [1, 1],
+        [1, 2],
+        [1, 3],
+        [1, 4],
+    ]
+
+    result = read_uncertainty_propagation_results(
+        metadata=metadata, results_key="temperature@trend_id_1"
+    )
+    assert np.all(
+        result.realization_output[0]
+        == np.array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7], dtype=np.float32)
+    )
+    assert np.all(
+        result.realization_output[-1]
+        == np.array([4.1, 4.2, 4.3, 4.4, 4.5, 4.6, 4.7], dtype=np.float32)
+    )
+    assert np.all(
+        result.mean_result
+        == np.array([2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7], dtype=np.float32)
+    )
+    assert np.allclose(
+        result.std_result,
+        np.array([1.41, 1.41, 1.41, 1.41, 1.41, 1.41, 1.41], dtype=np.float32),
+        rtol=0.01,
+    )
+
+    result = read_uncertainty_propagation_results(
+        metadata=metadata, results_key="absolute_pressure@trend_id_1"
+    )
+    assert np.all(
+        result.realization_output[0]
+        == np.array([10.1, 10.2, 10.3, 10.4, 10.5, 10.6, 10.7], dtype=np.float32)
+    )
+    assert np.all(
+        result.realization_output[-1]
+        == np.array([14.1, 14.2, 14.3, 14.4, 14.5, 14.6, 14.7], dtype=np.float32)
+    )
+    assert np.all(
+        result.mean_result
+        == np.array([12.1, 12.2, 12.3, 12.4, 12.5, 12.6, 12.7], dtype=np.float32)
+    )
+    assert np.allclose(
+        result.std_result,
+        np.array([1.414, 1.414, 1.414, 1.414, 1.414, 1.414, 1.414], dtype=np.float32),
+        rtol=0.001,
+    )
```

Comment:
Maybe you could use a data regression check here.

Reply:
Best use `num_regression`, which has a tolerance to compare floats.

Reply:
As you wish.